src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -324,14 +527,22 @@ namespace winrt::TerminalApp::implementation
         // appear first in the list. The ordering will be determined by the
         // match weight produced by _getWeight.
         std::priority_queue<WeightedCommand> heap;
-        for (auto action : _allActions)
+
+        // I hate this but this adds an "inOrderCounter" to allow commands with
+        // the same names and weights to be ordered in the order they were pushed
+        // onto the heap. This specifically helps out tab actions in the Tab Switcher,
+        // since it's quite common to have tab titles be the same.
```

Comment:
nit: file a followup and r4eplace "I hate this" with "TODO GH#THE_TEAM_HATES_THIS` :smile:

Reply:
gottem: #7205